src/drafts/SelectPanel2/SelectPanel.tsx

```diff
@@ -23,16 +24,16 @@ import {useFocusZone} from '../../hooks/useFocusZone'
 
 const SelectPanelContext = React.createContext<{
   title: string
-  description: string
+  description?: string
   panelId: string
   onCancel: () => void
   onClearSelection: undefined | (() => void)
   searchQuery: string
-  setSearchQuery: () => void
+  setSearchQuery: React.Dispatch<React.SetStateAction<string>>
```

Comment:
Are we defining its type here? 🤔  As far as I understand, we set the default values for the context here but this seems like doing something else or do I miss something?

Reply:
Yes, we can do both here.

`React.createContext<ContextTypes>(defaultValue)`

Reply:
Broadly speaking, how would you two feel about borrowing the `createContext` pattern from radix to help out with context usage like this across PRC? https://github.com/radix-ui/primitives/blob/main/packages/react/context/src/createContext.tsx#L3

I think the tl;dr is that this pattern has the context types itself be nullable (e.g. `ContextValueType | null`) but they leverage a hook to enforce that the value is always defined so they don't have to deal with default values.

Reply:
>  the context types itself be nullable (e.g. ContextValueType | null) but they leverage a hook to enforce that the value is always defined so they don't have to deal with default values.

That's cool! Can I pass on this PR and try it in a follow up PR?